cluster/addons/antrea/linux/daemonset.yaml

```diff
@@ -0,0 +1,208 @@
+apiVersion: apps/v1
+kind: DaemonSet
```

Comment:
/cc @jkh52 
There are concerns with the total number of DaemonSets we are running.

Reply:
Also the commit references windows but I notice this says linux in the path.

Reply:
I repurposed this to support linux as well. @jkh52 what is the concern about the number of DaemonSets?

Reply:
In my experience with DaemonSets, when there are enough nodes (example: hundreds or more) and HA clusters (3 masters) the proxy-server could have runaway memory ballooning (at times like rolling master restarts, when we expect many agent grpc connections). I found the root cause to be: the agent authentication path in proxy-server was getting throttled by client-go and accumulating resources.

I found stability improvements by tuning:
- proxy-agent syncinterval
- proxy-server k8s client throttling (https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/192)

Reply:
Are there node startup scripts that this could be placed in? Or is this solving a chicken/egg problem where Antrea network fabric is not known to be selected before the node boots?